aidb/vector_database/marqo_vector_database.py

```diff
@@ -0,0 +1,223 @@
+import uuid
+from dataclasses import dataclass, field
+import numpy as np
+import pandas as pd
+import marqo
+from typing import Dict,  Optional
+
+from aidb.utils.logger import logger
+from aidb.vector_database.vector_database import VectorDatabase
+
+@dataclass
+class MarqoAuth:
+  """
+  :param url: marqo url
+  :param username: marqo username
+  :param pwd: marqo password
+  :param api_key: marqo api key, user should choose input either username/pwd or api_key
+  """
+  url: Optional[str] = field(default=None)
+  username: Optional[str] = field(default=None)
+  pwd: Optional[str] = field(default=None)
+  api_key: Optional[str] = field(default=None)
+
+class MarqoVectorDatabase(VectorDatabase):
+  def __init__(self, marqo_auth: MarqoAuth, embedding_dim: int) -> None:
```

Comment:
Does Marqo support using GPU to accelerate the search process?

Reply:
Yes i guess, docs here - https://docs.marqo.ai/1.5.0/Guides/using_marqo_with_a_gpu/. I would go through this and implement.